web/store/chub.ts

```diff
@@ -0,0 +1,124 @@
+import { chubOptions } from '../pages/Chub/Chub'
+import { ChubItem } from '../pages/Chub/ChubItem'
+import { chubPage } from '../pages/Chub/ChubNavigation'
+import { createStore } from './create'
+
+export type ChubOptions = {
+  nsfw: boolean
+  tags: string
+  excludeTags: string
+  sort: string
+  search: string
+}
+
+export type ChubItem = {
+  name: string
+  description: string
+  fullPath: string
+}
+
+type ChubState = {
+  initLoading: boolean
+  chars: ChubItem[]
+  books: ChubItem[]
+  chubOptions: ChubOptions
+}
+
+export const CHUB_URL = `https://api.characterhub.org/api`
+
+const initState: ChubState = {
+  initLoading: true,
+  chars: [],
+  books: [],
+  chubOptions: {
+    nsfw: false,
+    tags: '',
+    excludeTags: '',
+    sort: 'Download Count',
+    search: '',
+  },
+}
+
+export const chubStore = createStore<ChubState>(
+  'chub',
+  initState
+)((_) => {
+  return {
+    async getChubChars() {
+      let sort = ''
+      switch (chubOptions.sort) {
```

Comment:
I'd break this out to a top-level function.
The `chubOptions` seems suitably placed in this store.